docs(glossary): Add more info about lossy/lossless compression (#32547)
mdn/content: files/en-us/glossary/lossless_compression/index.md

{{GlossarySidebar}}

**Lossless compression** is a class of data compression algorithms that allows the original data to be perfectly reconstructed from the compressed data. Lossless compression methods are reversible. Examples of lossless compression include {{glossary("GZip_compression", "gzip")}}, {{glossary("Brotli_compression", "brotli")}}, {{glossary("Zstandard compression", "Zstandard")}}, {{glossary("WebP")}}, and {{glossary("PNG")}}.

{{glossary("Lossy compression")}}, on the other hand, uses inexact approximations by discarding some data from the original file, making it an irreversible compression method.

> Some compression methods such as {{glossary("WebP")}} are capable of both lossy and lossless compression depending on their usage and this compression mode is typically user-configurable.

Reviewer: It seems slightly ambiguous what you mean by "this" in "this compression mode". Do you mean something like:

```suggestion
Some compression methods such as {{glossary("WebP")}} are capable of both lossy and lossless compression depending on their usage; the compression method used is typically user-configurable.
```

?

Reviewer: Yup, sounds much better
